src/schema-generation/root-types-generator.ts

```diff
@@ -23,80 +23,102 @@ import { UniqueFieldArgumentsGenerator } from './unique-field-arguments-generato
 import { UpdateInputTypeGenerator } from './update-input-types';
 
 export class RootTypesGenerator {
-    constructor(private readonly options: SchemaOptions = {}) {}
+    private readonly enumTypeGenerator: EnumTypeGenerator;
```

Comment:
There are many more places in the code where a field is not initialized in the constructor but directly on the field.

Are those occurrences any different than the ones that you fixed here?

Or did you only fix those places that lead to an error after enabling the tsconfig flag?

Reply:
The change is that now, field initializers run before the constructor, whereas they formerly ran after the constructor. So I only changed the places where field initializers depended on something set in the constructor.

I looked through all the files in the schema-generation directory, but not through the whole codebase, assuming the tests would catch other places.

What I don't understand is why this was not a typescript error (this.options was read before it was assigned). Maybe a newer typescript version would have caught it? I'll try

Reply:
it's actually [reported as an error since typescript 5.2](https://www.typescriptlang.org/play/?useDefineForClassFields=true&target=9&ts=5.2.2&ssl=4&ssc=16&pln=3&pc=5#code/MYGwhgzhAEAqCmEAu0DeAoaXrAPYDtkAnAV2CVyIAoAHIgSwDcwl5owAuaY+-Ac2gAfaCXwATeADNe8MQEpUAX0zYVWAEbQAvNCQALehAB0YANzp0ioA), so I updated to that version to be sure. It reported one more error, but I think that was actually a false-positive: https://github.com/AEB-labs/cruddl/pull/319/files#diff-fedcc11a3bb03b41cdf72a7d31a0ba1c598670339969cabe594d50bdff8db6d5 (does not make sense that static fields would be initialized after instance fields, and at runtime, it also was already initialized), but there was no harm in swapping the lines to fix the error